### ISSUE TITLE
Port "CustomPixiComponent" functionality from react-pixi

### ISSUE THREAD

**Issue by J00nz** [michalochman/react-pixi-fiber#19]
### Description
As a developers/users of the framework, it would be great if the "CustomPixiComponent" got ported to react-pixi-fiber. 

### Steps to reproduce

1. Start porting your solution from react-pixi to react-pixi-fiber
2. Get annoyed that you cant use your "CustomPixiComponents".

**Comment by J00nz** [comment]
https://github.com/michalochman/react-pixi-fiber/pull/20

**Comment by michalochman** [comment]
Hey @J00nz, can you show me an example of such `CustomPixiComponent`?

**Comment by J00nz** [comment]
When dealing with custom PIXI types, such as https://github.com/pixijs/pixi-spine or other third-party custom native PIXI components, having the ability to add custom-components and wrap its functionality is something that we've sorely needed.
If you have another way of achieving this we are of course open to suggestions...

**Comment by michalochman** [comment]
Can you show me an example of how you use it in [`react-pixi`](https://github.com/Izzimach/react-pixi)? As far as I know, `CustomPIXIComponent` doesn't work in `react-pixi`, here's the issue I opened over a year ago: https://github.com/Izzimach/react-pixi/issues/72.

Aside from that, I am currently using `pixi-spine` without any need of custom component, you can just do it through regular React component composition, e.g. something like:

```javascript
class SpineAnimation extends React.Component {
  componentDidMount() {
    // initialize Spine here
    this.spine = new PIXI.spine.Spine(/* pass Spine data here */)

    // add Spine to Container
    this.container.addChild(this.spine)
  }
  
  render() {
    return (
      <ReactPIXI.DisplayObjectContainer ref={c => this.container = c} />
    )
  }
}
```

Then you can implement any functionality you need by using `this.spine` reference.

Is your solution with custom component even more simple?

**Comment by J00nz** [comment]
I would say that it works in a similar manner but provides a leaner interface, since it doesn't force the users to add unnecessary wrapper components. Plus it further simplifies the porting of existing react-pixi applications. 
Plus I also added some basic resizing and ported the way react-pixi handles pixi point properties.

**Comment by olivoil** [comment]
Does this work if you need a custom Container type with sub-component from react-pixi-fiber ? For example to add https://github.com/davidfig/pixi-viewport ?

**Comment by J00nz** [comment]
@olivoil No, not in its current implementation, it merely functions as a glue between native pixi.js and react-pixi-fiber.

@michalochman rewriting your previous example to CustomPixiComponent:
```
const SpineAnimation = CustomPixiComponent({
  customDisplayObject(props) {
    return new PIXI.spine.Spine(props.spine.spineData)
  }
}
```

**Comment by michalochman** [comment]
@J00nz thanks, is this an implementation in react-pixi or the implementation in react-pixi-fiber using #20? Are you actually using `customDidAttach` and `customWillDetach`? I would like to properly plan development of this feature.

**Comment by J00nz** [comment]
@michalochman the example works in both.
We've found it necessary to use at least `customWillDetach`, since it provides a way to do cleanup of variables etc that might exist to prevent memory leaks. `customDidAttach` we haven't used much, I just added it to https://github.com/michalochman/react-pixi-fiber/pull/20 to ensure full compatibility with existing applications.

**Comment by J00nz** [comment]
```
const SpineAnimation = CustomPixiComponent({
  customDisplayObject(props) {
    window.addEventListener('resize', DoingResizeStuff);
    return new PIXI.spine.Spine(props.spine.spineData)
  }
  customWillDetach() {
    window.removeEventListener('resize', DoingResizeStuff);
  }
}
```

**Comment by michalochman** [comment]
Hey @J00nz is there a chance we can talk outside github? I've created a gitter room here: https://gitter.im/react-pixi-fiber.

**Comment by michalochman** [comment]
Fixed in #25
